exercicios/para-casa/Sayonara/SAYONARA.sistema_de_alunas.py

```diff
@@ -0,0 +1,152 @@
+from dataset_alunas import dataset
+
+def main():
+    print("\n---  Seja bem vinda a Escola do Reprograma!  ---")
+    print("Sistema de informações de alunas")
+    
+    while True:
+        cod_opcao = obter_opcao()
+        
+        if cod_opcao == 1: incluir_nova_aluna()
+        elif cod_opcao == 2: consultar_lista_alunas(dataset)
+        elif cod_opcao == 3: consultar_faltas_aluna()
+        elif cod_opcao == 4: consultar_notas_aluna()
+        elif cod_opcao == 5: consultar_status_aprovacao()
+        elif cod_opcao == 6: print("Encerrando o programa..."); break
+
+def obter_opcao():
+    codigo_opcao = 0
+
+    while codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+        try:
+            codigo_opcao = int(input("\nEscolha uma opção:\n"
+                                    "1 - Incluir uma nova aluna\n"
+                                    "2 - Consultar lista de alunas\n"
+                                    "3 - Consultar faltas da aluna\n"
+                                    "4 - Consultar notas da aluna\n"
+                                    "5 - Consultar status de aprovação\n"
+                                    "6 - Sair do sistema\n"
+                                    "Opção: "))
+                
+            if codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+                print("Opção inválida. Por favor, escolha uma opção válida (1 a 5).\n")
+        except ValueError:
+            print("Entrada inválida. Por favor, digite um número inteiro.\n")
+            
+        return codigo_opcao
+    
+def incluir_nova_aluna():
+    print("Informe os dados ")
+    nome= input("Informe o nome e o sobrenome :")
+    turma = input("informe a turma :" )
+    notas = pedir_notas()
+    nota_participaçao=float(input("participação da aluna: "))
+    presença= chamada() 
+    dataset[(nome.split()[0] , nome.split()[1])] = {
+        "Turma": turma,
+         "Notas": notas,
+         "Presença": presença,
+         "Paticipaçao": nota_participaçao
+    }
+    print("Aluna cadastrada com sucesso ")
+    pass
+    #TODO - Implentar a função
+def pedir_notas():
+   #quantidade_notas=input("informa a quantidade das notas:")
+   notas=[]
+   for i in range(3):
+         nota=float(input(f"Digite a {i+1}° nota:" ))
+         notas.append(nota)
+
+
+   return notas 
+ 
+def chamada(): 
+    aulas=[] 
+    for  i in range (5): 
+     aula=int(input(f" Digite 1 se presente ou 0 se ausente, na aula :{i+1}°: presença:"))  
+    aulas.append(aula)
+    return aulas 
+
+    
+def consultar_lista_alunas(dataset):
+     print("Lista alunas ")
+     for aluna_tuple in dataset:
+        nome_c = aluna_tuple[ 0 ]  + " " + aluna_tuple[1]
+        nome =nome_c.strip('(').strip()
+        print (nome)
```

Comment:
Podemos pensar em um nome mais intuitivo para representar o nome e sobrenome das alunas. Nas tuplas, temos possibilidade de realizar o desempacotamento para nomear as variáveis. Assim, ajuda na legibilidade e clareza do código.

**Sugestão:**
```suggestion
    for nome, sobrenome in dataset:
        print(f"{nome} {sobrenome}")
```